### PR TITLE
TST: camper_program.py: tests and exception handling

### PR DESCRIPTION
**Pull request by westurner**
- This builds atop https://github.com/4dsolutions/python_camp/pull/3 .
- I gold-plated it for yahs. Branch-level coverage is a different metric than line-level coverage.
- Today I learned about `unittest.mock.patch` and stdin and stdout.
- This is at 82% coverage. The module docstring lists how to run the tests and check for line-level test coverage. There are a few outstanding tasks.
- Additional `Calculator` implementations could use e.g. NumPy, SymPy, your own implementations of exponentiation, etc. Having the tests helps to identify what will break. Having tests enables *"Fearless Refactoring"*: yeah, it's going to break the existing tests, but at least we know where.

### CODE REVIEW

File: camper_program.py [hunk 1]
```diff
@@ -3,105 +3,394 @@
...
+    def run_loop(self):
+        menu_options = self.get_menu_options()
+        looping = True
+        while looping:
+            for n, (menustr, _) in menu_options.items():
+                print(f"            {n}. {menustr}")
+            print("\n            Pick one please")
+
+            try:
+                do_it = input("    >>> ")
```
Comment (westurner) [inline camper_program.py:130]: I did not change this variable name.

https://en.wikipedia.org/wiki/Variable_(computer_science)#Naming_conventions

[Starsky & Hutch (3/5) Movie CLIP - Do It (2004) HD](https://www.youtube.com/watch?v=P9pa_8-WdlU)

**Comment by 4dsolutions** [comment]
Excellent patches supplied by Wes and applied during camp as part of a live demo.

**Comment by 4dsolutions** [comment]
Everything went smoothly.

**Comment by westurner** [comment]
Thanks!

**Pull request merged**